diff --git a/bin/vue-tsc.js b/bin/vue-tsc.js
index f0c7776ff9c469024ded582a5391c20e34de7d7b..c3ab5bdaa45585467b02ce239171f9d6db4ff2de 100644
--- a/bin/vue-tsc.js
+++ b/bin/vue-tsc.js
@@ -38,6 +38,10 @@ fs.readFileSync = (...args) => {
 			);
 		}
 
+		if (semver.gte(tsPkg.version, '5.1.6')) {
+			tryReplace(`Debug.assert(getRootLength(fromDirectory) > 0 === getRootLength(to) > 0, "Paths must either both be absolute or both be relative");`, ``);
+		}
+
 		return tsc;
 
 		function tryReplace(search, replace) {